### PR TITLE
Support for user auth().

### PR DESCRIPTION
**Pull request by mkucharz**
Not sure if it is right place...

### CODE REVIEW

File: syncano/models/accounts.py [hunk 1]
```diff
@@ -133,6 +137,18 @@ def reset_key(self):
...
         return connection.request('POST', endpoint)
 
+    def auth(self):
+        properties = self.get_endpoint_data()
+        endpoint = self._meta.resolve_endpoint('auth', properties)
+        connection = self._get_connection()
+
+        data = {
+            'username': connection.username,
+            'password' : connection.password
```
Comment (23doors) [inline syncano/models/accounts.py]: obsolete whitespace before `:`

File: syncano/models/accounts.py [hunk 2]
```diff
@@ -133,6 +137,18 @@ def reset_key(self):
...
+        }
+
+        return connection.request('POST', endpoint, data=data)
```
Comment (23doors) [inline syncano/models/accounts.py]: Maybe we should set user key automatically after auth?

**Comment by 23doors** [comment]
some test would be nice ;)

File: syncano/models/accounts.py (hunk 2 above)
Comment (opalczynski) [inline syncano/models/accounts.py]: :+1:

Comment (mkucharz) [inline syncano/models/accounts.py]: Yep, but there is other place to do that - this only simply to test credentials. For that, I think that we should refactor a little bit user auth in connection.py.

**Comment by mkucharz** [comment]
Now it can be used that way, we have to rethink how to do it in case we want to do it during the connection.

```
connection = syncano.connect(api_key='key', instance_name='picoblocks')
print connection.User().auth(username='maciej@kucharz.net', password='test123')
```

File: syncano/models/accounts.py [hunk 3]
```diff
@@ -133,6 +137,21 @@ def reset_key(self):
         connection = self._get_connection()
         return connection.request('POST', endpoint)
 
+    def auth(self, username=None, password=None):
+        properties = self.get_endpoint_data()
+        endpoint = self._meta.resolve_endpoint('auth', properties)
+        connection = self._get_connection()
+
+        if not (username and password):
+            raise SyncanoValueError('You need provide username and password!')
```
Comment (opalczynski) [inline syncano/models/accounts.py]: We usually use dots et the end of Exception msg.

**Comment by opalczynski** [comment]
@mkucharz what about simple test case?  I think that you can add it in integration_test_accounts:)

**Comment by opalczynski** [comment]
LGTM.

**Pull request merged**